feat(ignoreFlag): Update filtering ignore flag.

Derek-94 opened the pull request:
### Details

- User clicks ignore checkbox, data below renders differently.

### 구현내용

![mr camel01](https://user-images.githubusercontent.com/52649378/127635700-5fb2186e-7910-4b83-870b-4e2b5462b6c4.gif)

사용자가 박스를 클릭함에 따라 하단 데이터가 조정됩니다.

---

`ProductHistory` 컴포넌트에는 아래와 같은 변수들이 `state` 로 지정되어 있으며,

![image](https://user-images.githubusercontent.com/52649378/127651795-ee7caae3-a6c3-499c-b250-18e5805bd3b4.png)

- `productHistoryOrigin` : 원본데이터
- `productHistoryModified` : 사용자가 필터링 아무거나 건 순간, 편집된 데이터 배열을 가지는 변수

위 짤에 보이는 체크박스에는 아래 이벤트함수가 바인딩됩니다.

![image](https://user-images.githubusercontent.com/52649378/127651926-380a48fb-e198-4f3c-a24d-510d45578742.png)

- `state` 의 `ignoreFlag` 의 상태에 따라 분기문으로 처리하였습니다. 

![image](https://user-images.githubusercontent.com/52649378/127651976-1cf3b4d6-01db-4675-b085-c094255f1764.png)

- 맨 처음에는 81번째줄이 수행됩니다. 
- 하지만 사용자가 조작을 함에따라, 68번째 줄만 수행됩니다. -> 그때는 `productHistoryModified` 의 길이를 보고 81번째 줄을 막습니다.

closes #8

Derek-94 committed:
feat(ignore): Update filtering ignore flag.

- User clicks ignore checkbox, data below renders differently.

blingblin-g commented on src/Components/ProductHistory.jsx:
```diff
@@ -8,13 +8,31 @@ export default class ProductHistory extends Component {
+      });
+    } else {
+      this.setState({
+        productHistory: JSON.parse(localStorage.getItem('productHistory')),
```
매번 로컬스토리지에 접근해서 가져오는 것보다 원본을 state로 관리해서 props로 넘겨주는 게 효율적이지 않을까요?

Derek-94 replied:
수정할게요

blingblin-g commented on src/Components/ProductHistory.jsx:
```diff
@@ -48,11 +66,12 @@ export default class ProductHistory extends Component {
-            {productHistory.map(product => (
-              <div style={{ margin: '1rem 0' }}>
+            {productHistory.map((product, index) => (
+              <div key={index} style={{ margin: '1rem 0' }}>
```
이건 임시로 한거겠죠? 나중에 꼭 바꿔주세요

Derek-94 replied:
네

blingblin-g reviewed (commented):
코드리뷰했습니다.

Derek-94 committed:
refactor: Applied feedbacks.

- Not revise original arrays.
- Whenever user clicks filter, page renders productHistoryModified, not
  productHistoryOrign.

Derek-94 reviewed (commented):
수정하겠습니다.

blingblin-g reviewed (approved):
수고하셨습니다
